aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+
++-------------------------------+--------------------------------+
+|asp-validation-summary         |  Validation messages displayed |  
++===============================+================================+
+|ValidationSummary.All          | Property and model level       |
++-------------------------------+--------------------------------+  
+|ValidationSummary.ModelOnly    | Model                          |
++-------------------------------+--------------------------------+  
+|ValidationSummary.None         | None                           |
++-------------------------------+--------------------------------+  
+
+Sample
+^^^^^^^^^
+
+In the following example, the data model is decorated with ``DataAnnotation`` attributes, which generates validation error messages on the ``<input>`` element.  When a validation error occurs, the Validation Tag Helper displays the error message:
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterValidation.cshtml
+  :language: HTML
+  :emphasize-lines: 4,6,8
+  :lines: 1-10
+
+The generated HTML (when the model is valid):
+
+.. code-block:: HTML
+  :emphasize-lines: 2,3,8,9,12,13
+  
+  <form action="/DemoReg/Register" method="post">
+    <div class="validation-summary-valid" data-valmsg-summary="true">
+    <ul><li style="display:none"></li></ul></div>
+    Email:  <input name="Email" id="Email" type="email" value="" 
+     data-val-required="The Email field is required." 
+     data-val-email="The Email field is not a valid e-mail address." 
+     data-val="true"> <br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Email"></span><br>
+    Password: <input name="Password" id="Password" type="password" 
+     data-val-required="The Password field is required." data-val="true"><br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Password"></span><br>
+    <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The Select Tag Helper
+-------------------------
+
+- Generates `select <https://www.w3.org/wiki/HTML/Elements/select>`__ and associated `option <https://www.w3.org/wiki/HTML/Elements/option>`__ elements for properties of your model. 
+- Has an HTML Helper alternative ``Html.DropDownListFor`` and ``Html.ListBoxFor``
+
+The `Select Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/SelectTagHelper/index.html>`__ ``asp-for`` specifies the model property  name for the `select <https://www.w3.org/wiki/HTML/Elements/select>`__ element  and ``asp-items`` specifies the `option <https://www.w3.org/wiki/HTML/Elements/option>`__ elements.  For example:
```

Comment:
Something is missing before "`asp-for`". Perhaps sentence should be more like "The **Select Tag Helper**'s `asp-for` attribute ... and its `asp-items` attribute ..."?